src/handlers/int-test-support/helpers/testDataHelper.ts

```diff
@@ -39,36 +38,29 @@ export const getVendorServiceAndRatesFromConfig =
     return testDataRetrievedFromConfig;
   };
 
-export const generateTransactionEventsViaFilterLambda = async (
-  eventTime: string,
-  transactionQty: number,
-  eventName: string
-): Promise<string[]> => {
-  const eventIds: string[] = [];
-  for (let i = 0; i < transactionQty; i++) {
-    const updatedSQSEventPayload = await updateSQSEventPayloadBody(
-      eventTime,
-      eventName
-    );
+export interface GenerateEventsResult {
+  success: boolean;
+  eventId?: string;
+}
+
+export const generateEventViaFilterLambdaAndCheckEventInS3Bucket = async (
+  payload: EventPayload
+): Promise<GenerateEventsResult> => {
+  try {
+    const updatedSQSEventPayload = await updateSQSEventPayloadBody(payload);
     const functionName = `${resourcePrefix()}-filter-function`;
     await invokeLambda({ functionName, payload: updatedSQSEventPayload });
     const json = JSON.parse(updatedSQSEventPayload);
     const eventId = JSON.parse(json.Records[0].body).event_id;
-    await poll(
-      async () =>
-        await listS3Objects({
-          bucketName: `${resourcePrefix()}-storage`,
-          prefix: `btm_transactions`,
-        }),
-      (result) => !!result?.Contents?.some((data) => data.Key?.match(eventId)),
-      {
-        nonCompleteErrorMessage:
-          "event id not appeared in s3 within the given timeout",
-      }
-    );
-    eventIds.push(eventId);
+    const eventExistsInS3 = await checkS3BucketForEventId(eventId, 7000);
+    if (!eventExistsInS3) {
+      return { success: false };
```

Comment:
If you return the eventID from invalid payloads then, they can also be used in the tests. I tried this with the suggestion I made for the athena-query-tests.ts
```suggestion
            return { success: false, eventId };
```

Reply:
done